### PR TITLE
Don't keep mutable config instance on a field

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Don't keep mutable config instance on a field

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java [hunk 1]
```diff
@@ -467,7 +481,25 @@ public LocalExecutionPlanner(
...
+        this.largePartitionedRangeRowLimitPerDriver = dynamicFilterConfig.getLargePartitionedRangeRowLimitPerDriver();
+        this.smallBroadcastRangeRowLimitPerDriver = dynamicFilterConfig.getSmallBroadcastRangeRowLimitPerDriver();
+        this.smallPartitionedRangeRowLimitPerDriver = dynamicFilterConfig.getSmallPartitionedRangeRowLimitPerDriver();
+        this.largeBroadcastMaxSizePerOperator = dynamicFilterConfig.getLargeBroadcastMaxSizePerOperator();
+        this.largePartitionedMaxSizePerOperator = dynamicFilterConfig.getLargePartitionedMaxSizePerOperator();
+        this.smallBroadcastMaxSizePerOperator = dynamicFilterConfig.getSmallBroadcastMaxSizePerOperator();
+        this.smallPartitionedMaxSizePerOperator = dynamicFilterConfig.getSmallPartitionedMaxSizePerOperator();
+
+
+        this.largePartitionedMaxDistinctValuesPerDriver = dynamicFilterConfig.getLargePartitionedMaxDistinctValuesPerDriver();
```
Comment (ksobolew) [inline core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java:502]: Is this gap in purpose?

**Review by ksobolew — APPROVED** [review]
❤️ 🙌

File: core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java (hunk 1 above)
Comment (ebyhr) [reply core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java]: No. Thanks for catching that. Removed the empty lines.

**Comment by ebyhr** [comment]
CI hit #14487

File: core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java [hunk 2]
```diff
@@ -119,7 +119,9 @@ public DynamicFilterService(Metadata metadata, FunctionManager functionManager,
         this.metadata = requireNonNull(metadata, "metadata is null");
         this.functionManager = requireNonNull(functionManager, "functionManager is null");
         this.typeOperators = requireNonNull(typeOperators, "typeOperators is null");
-        this.dynamicFilterConfig = requireNonNull(dynamicFilterConfig, "dynamicFilterConfig is null");
+        requireNonNull(dynamicFilterConfig, "dynamicFilterConfig is null");
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java:122]: redundant

File: core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java [hunk 3]
```diff
@@ -467,7 +481,23 @@ public LocalExecutionPlanner(
         this.joinCompiler = requireNonNull(joinCompiler, "joinCompiler is null");
         this.operatorFactories = requireNonNull(operatorFactories, "operatorFactories is null");
         this.orderingCompiler = requireNonNull(orderingCompiler, "orderingCompiler is null");
-        this.dynamicFilterConfig = requireNonNull(dynamicFilterConfig, "dynamicFilterConfig is null");
+        requireNonNull(dynamicFilterConfig, "dynamicFilterConfig is null");
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java:484]: redundant, see https://github.com/trinodb/trino/pull/13940

**Review by findepi — APPROVED** [review]

**Comment by ebyhr** [comment]
Compatibility failed due to lack of the latest docker image. https://trinodb.slack.com/archives/CFP480UKX/p1665108934638869

**Pull request merged**